src/mint.rs

```diff
@@ -108,11 +112,71 @@ impl ReissueTransaction {
     }
 
     fn validate_balance(&self) -> Result<()> {
-        let input: u64 = self.inputs.iter().map(|input| input.amount()).sum();
-        let output: u64 = self.outputs.iter().map(|output| output.amount).sum();
+        // Calculate sum(input_commitments) and sum(output_commitments)
+        let inputs: RistrettoPoint = self
+            .inputs
+            .iter()
+            .map(|input| {
+                input
+                    .content
+                    .commitment
+                    .decompress()
+                    .ok_or(Error::AmountCommitmentInvalid)
+            })
+            .sum::<Result<RistrettoPoint, _>>()?;
+        let outputs: RistrettoPoint = self
+            .outputs
+            .iter()
+            .map(|output| {
+                output
+                    .commitment
+                    .decompress()
+                    .ok_or(Error::AmountCommitmentInvalid)
+            })
+            .sum::<Result<RistrettoPoint, _>>()?;
+
+        // Verify the range proof for each output.  (bulletproof)
+        // This validates that the committed amount is a positive value.
+        // (somewhere in the range 0..u64::max)
+        let bullet_gens = BulletproofGens::new(64, 1);
+        let ped_commits = PedersenGens::default();
```

Comment:
rename to `pc_gen` since we are using this for it's knowledge of the base pederson points.

This generator is only used for it's two curve points (H and G)

Reply:
sure, yeah that came from bulletproofs sample code.